### PR TITLE
SNOW-2467769: Add binary feature tests

### PR DESCRIPTION
**Pull request by sfc-gh-rkowalski**
Created new `BinaryTests.java` and `BinaryLobTests.java` files containing comprehensive JDBC test cases for binary type functionality. The tests cover basic type handling with statement and table operations (insert, update, delete, select), edge cases with NULL values and special characters, and complex scenarios with binary LOB values.

### CODE REVIEW

File: jdbc/src/test/java/net/snowflake/jdbc/e2e/types/BinaryTests.java [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+    return String.format("%0" + SEQUENTIAL_BINARY_TEXT_WIDTH + "d", value)
+        .getBytes(StandardCharsets.UTF_8);
+  }
+
+  private static byte[] bytesFromHex(String hex) {
+    int length = hex.length();
+    byte[] bytes = new byte[length / 2];
+    for (int i = 0; i < length; i += 2) {
+      int high = Character.digit(hex.charAt(i), 16);
+      int low = Character.digit(hex.charAt(i + 1), 16);
```
[the remark this answers is not in the document]
Comment (sfc-gh-rkowalski) [reply jdbc/src/test/java/net/snowflake/jdbc/e2e/types/BinaryTests.java:563]: This is just test util where we control the input.

File: jdbc/src/test/java/net/snowflake/jdbc/e2e/types/BinaryTests.java [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+    execute(
+        connection,
+        "INSERT INTO "
+            + tableName
+            + " VALUES (1, X''), (2, X'00'), (3, X'FF'), (4, X'000000'), (5, X'48006500')");
+
+    // When Query "SELECT * FROM {table} ORDER BY 1" is executed
+    withQueryResult(
+        connection,
+        "SELECT col FROM " + tableName + " ORDER BY id",
```
[the remark this answers is not in the document]
Comment (sfc-gh-rkowalski) [reply jdbc/src/test/java/net/snowflake/jdbc/e2e/types/BinaryTests.java]: This is intended

**Review by sfc-gh-merbel — APPROVED** [review]
Please solve copilot comments

**Review by sfc-gh-pfus — APPROVED** [review]

**Review by sfc-gh-dheyman — APPROVED** [review]

**Comment by sfc-gh-rkowalski** [comment]
### Merge activity

* **Mar 17, 9:24 AM UTC**: The merge label 'merge_queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) once it meets the requirements.
* **Mar 17, 9:24 AM UTC**: `sfc-gh-rkowalski` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver).
* **Mar 17, 9:28 AM UTC**: CI is running for this pull request on a draft pull request ([#528](https://app.graphite.com/github/pr/snowflakedb/universal-driver/528)) due to your merge queue CI optimization settings.
* **Mar 17, 9:39 AM UTC**: CI is running for this pull request on a draft pull request ([#530](https://app.graphite.com/github/pr/snowflakedb/universal-driver/530)) due to your merge queue CI optimization settings.
* **Mar 17, 10:05 AM UTC**: This pull request has been added to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) failure handling queue because the batch it was included in failed to pass CI.
* **Mar 17, 10:05 AM UTC**: CI is running for this pull request on a draft pull request ([#531](https://app.graphite.com/github/pr/snowflakedb/universal-driver/531)) due to batch failure handling.
* **Mar 17, 10:30 AM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI (rust-core-status)).
* **Mar 17, 10:30 AM UTC**: [PR #469](https://app.graphite.com/github/pr/snowflakedb/universal-driver/469) has been removed from the merge queue after undergoing failure handling due to failing CI checks: rust-core-status. Check for failures in draft pull request ([#531](https://app.graphite.com/github/pr/snowflakedb/universal-driver/531)).
* **Mar 17, 12:48 PM UTC**: The merge label 'merge_queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) once it meets the requirements.
* **Mar 17, 12:48 PM UTC**: `sfc-gh-rkowalski` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver).
* **Mar 17, 12:51 PM UTC**: CI is running for this pull request on a draft pull request ([#538](https://app.graphite.com/github/pr/snowflakedb/universal-driver/538)) due to your merge queue CI optimization settings.
* **Mar 17, 1:15 PM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI (rust-core-status)).
* **Mar 17, 1:58 PM UTC**: The merge label 'merge_queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) once it meets the requirements.
* **Mar 17, 1:58 PM UTC**: `sfc-gh-rkowalski` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver).
* **Mar 17, 2:02 PM UTC**: CI is running for this pull request on a draft pull request ([#547](https://app.graphite.com/github/pr/snowflakedb/universal-driver/547)) due to your merge queue CI optimization settings.
* **Mar 17, 2:31 PM UTC**: CI is running for this pull request on a draft pull request ([#551](https://app.graphite.com/github/pr/snowflakedb/universal-driver/551)) due to your merge queue CI optimization settings.
* **Mar 17, 3:09 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=snowflakedb&repo=universal-driver) via draft PR: [#551](https://app.graphite.com/github/pr/snowflakedb/universal-driver/551).
